extension/source/Home/Wallet/Wallets/TransactionsTable.tsx

```diff
@@ -0,0 +1,213 @@
+import * as React from 'react';
+import {
+  useReactTable,
+  getCoreRowModel,
+  getPaginationRowModel,
+  flexRender,
+  ColumnDef,
+} from '@tanstack/react-table';
+import {
+  ShareNetwork,
+  CaretLeft,
+  CaretRight,
+  ArrowUpRight,
+  UserCircle,
+} from 'phosphor-react';
+
+import { ethers } from 'ethers';
+import onAction from '../../../helpers/onAction';
+import DisplayNonce from './DisplayNonce';
+import useCell from '../../../cells/useCell';
+import { useQuill } from '../../../QuillContext';
+import Loading from '../../../components/Loading';
+import { QuillTransaction } from '../../../types/Rpc';
+import formatCompactAddress from '../../../helpers/formatCompactAddress';
+
+interface ITransactionsTable {
+  selectedAddress: string;
+}
+
+export const TableHeader: React.FunctionComponent<ITransactionsTable> = ({
+  selectedAddress,
+}) => {
+  return (
+    <div className="flex justify-between p-6">
+      <div className="flex gap-4">
+        <div className="flex gap-2 items-center">
+          <UserCircle className="text-blue-500 icon-md" />
+          <div className="">
+            <span className="text-blue-500">From :</span>{' '}
+            {formatCompactAddress(selectedAddress)}
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const TransactionsTable: React.FunctionComponent<ITransactionsTable> = ({
+  selectedAddress,
+}) => {
+  const quill = useQuill();
+  const transactions = useCell(quill.cells.transactions);
+  const data =
```

Comment:
Is the idea that we would eventually query the data on chain to get transactions or will we always just look at local storage?

Reply:
For now we can just look at the local transactions as the on-chain tx could include txs from more than 1 wallet as part of batching. It would be a little bit difficult to parse the individual actions per wallet but certainly is doable and could be worked upon in future! This will be particularly useful when the user is migrating keys from either a burner wallet or changing browsers.

Reply:
Nice that makes sense.  Thanks for the explanation!